pubpub/pubpub
# feature: code blocks w/ plugin

We are going with this approach!

_Test Plan_
create code block in a pub and play with them generally. Steps below. NB: the code blocks behave as a text block, which is to say, they are not deletable if they are the only text block in the document just as the only `paragraph` block would not be deletable. We can change this behavior, but it would be a departure from block behavior to accommodate image-block-like behavior.
Make sure to hit each of the below:
+ create a code block by typing three backticks to enter it
+ create a code block by clicking the lego-block-shaped codeblock toggle button in the formatting bar
+ create a code block by selecting `code` from the dropdown at the left edge of the formatting bar
+ exit the codeblock by pressing `return` from an empty line in the codeblock
+ exit the codeblock by navigating to the right from the last character (per NB above, this is impossible if there isn't a following text block to navigate into)
+ while inside a codeblock, select different language syntax highlighting grammars, then reload the page
+ copy + paste into/out of codeblocks
+ delete codeblocks with backspace (provided there is a text node above the codeblock in the document)

TODO:
+ implement the language selector dropdown component using Blueprint components for visual consistency, using the `createSelect` api from the vendor module.
+ PDF export
